src/NetscriptFunctions/Singularity.ts

```diff
@@ -95,6 +95,12 @@ export function NetscriptSingularity(): InternalAPI<ISingularity> {
     getOwnedSourceFiles: () => () => {
       return [...Player.sourceFiles].map(([n, lvl]) => ({ n, lvl }));
     },
+    getAugmentationFactions: (ctx) => (_augName) => {
+      helpers.checkSingularityAccess(ctx);
+      const augName = getEnumHelper("AugmentationName").nsGetMember(ctx, _augName);
+      const aug = Augmentations[augName];
+      return aug.factions;
```

Comment:
Don't return the real array directly to the player. When we pass game information to the player in the form of an object/array/etc, we need to give them a copy of the real thing so any modifications the player makes to the return value do not impact the game.

```suggestion
      return aug.factions.slice();
```